### PR TITLE
Fix a bug in expand

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
Ready for review

**Comment by bjodah** [comment]
+1 Looks right, and verified independently using SymPy.

**Comment by isuruf** [comment]
Thanks for the review

File: symengine/tests/basic/test_arit.cpp [hunk 1]
```diff
@@ -1376,6 +1376,10 @@ TEST_CASE("Expand2: arit", "[arit]")
     r1 = add(x, mul(i2, add(y, z)));
     r2 = expand(r1, false);
     REQUIRE(eq(*r1, *r2));
+
+    r1 = expand(pow(add(one, mul(i2, sqrt(add(y, z)))), i2));
+    r2 = expand(add(one, mul(i4, add(y, add(z, sqrt(add(y, z)))))));
+    REQUIRE(eq(*r1, *r2));
```
Comment (certik) [inline symengine/tests/basic/test_arit.cpp]: I wonder if we could start using the `Expression` in tests, to make them more readable (possibly slower, but we don't care about that in tests).

**Comment by certik** [comment]
How is performance affected?

File: symengine/tests/basic/test_arit.cpp (hunk 1 above)
Comment (isuruf) [reply symengine/tests/basic/test_arit.cpp]: Yes, that's a good idea. That way, we can add to functionality in `Expression` if they are not added.

**Comment by isuruf** [comment]
No visible affect. There's a fast code path for multiplication by `1`, so the performance shouldn't be affected.

**Comment by certik** [comment]
I tested the expand2 benchmark, and didn't see an effect either.

**Pull request merged**